current/runtime/floats.md

```diff
@@ -4,9 +4,11 @@ lang: en
 title: Floating Point Numbers
 ---
 
-When developing for the Substrate runtime and doing calculations it is appealing to use floating point numbers for e.g. representing fractions. This document covers why traditional floating point numbers are problematic in a runtime context and what alternatives exist.
+When developing for the Substrate runtime and doing calculations it is appealing to use floating point numbers for e.g. representing fractions. In Substrate it is important that each node arrives at the same deterministic result for consensus. This document covers why traditional floating point numbers are problematic in runtime and what alternatives exist.
 
-## Problems with Floating Point Numbers in the Runtime
+## Problems with Floating Point Numbers in Runtime
+
+The standard way of handling floating points is dependent on the need for decimal places by using one of the in-built primitive types of handlingfix point arithmetic. Floating point numbers are not deterministic and fixed point computation is safe for Substrate since it represents all rationals as a fraction thus resolving a deterministic result. The two types which are used to handle large fixed point arithmetic are Permill and Perbill, you can refer to [safe math](https://substrate.dev/recipes/3-entrees/safemath.html)for further information and implementation.
```

Comment:
```suggestion
Floating point numbers are not deterministic and thus cannot be used to run a deterministic block production algorithm. The standard way of handling rational (or real) numbers depends on how many decimal places are needed and then choosing one of the in-built primitive types in Substrate providing fixed point arithmetic. Fixed point computation is safe for Substrate since it represents all rationals as a fraction thus resolving to a deterministic result. The two types which are used to handle large fixed point arithmetic are `Permill` and `Perbill`, you can refer to [safe math](https://substrate.dev/recipes/3-entrees/safemath.html) for further information and implementation.
```

Reply:
A few comments:
+ AFAICT This paragraph does not seem to focus on the problems with floating point numbers, so you might want to tease those things apart?
+ This is a knowledge base article so I'm not sure we want to recommend `Permill` and `Perbill` straight away, but instead explain the general/transferable reasoning.
+ I think people (including myself) easily confuse and conflate floats with rational/real numbers when they are distinct. So I would want to avoid using "floats" in places that refer to real/rational numbers. About which words to use for describing this: IMO float (floating point number), rational (rational number) and fixed point number are all seperate concepts. I, personally, would try to always use them in a way that distinguishes them from each other and makes sure that they remain distinct in the context of this article:
  + rational numbers are a mathematical concept of numbers that can be represented by the quotient of 2 integers https://en.wikipedia.org/wiki/Rational_number
  + floats are an encoding (for real numbers, including rational numbers) with limited precision
  + fixed point numbers are a different encoding also with limited precision

Reply:
got it -- 
focus on floating point number issues. 
explain general/transferable reasons vs 'permill' 
possibly define rational/real numbers when referring to floats